src/libraries/System.Private.CoreLib/src/System/Guid.cs

```diff
@@ -71,6 +71,11 @@ public Guid(ReadOnlySpan<byte> b)
             _h = b[12];
             _i = b[13];
             _j = b[14];
+
+            static void ThrowArgumentException()
```

Comment:
```suggestion
            [StackTraceHidden]
            static void ThrowArgumentException()
```
in order to keep the previous stacktrace.

Reply:
Previously we'd see the following stack trace in case of an exception:
```
Guid..ctor(Span)
MethodCallingGuid
```
With @DaZombieKiller's change it's now:
```
Guid.ThrowArgumentException
MethodCallingGuid
```
with the attribute you suggest, the stack trace will be:
```
MethodCallingGuid
```

Reply:
Addressed in https://github.com/dotnet/runtime/pull/78446/commits/057f68cd888dca03b405fb1f9d7bb93c4c241963.

Reply:
Why this particular throw helper and not others?